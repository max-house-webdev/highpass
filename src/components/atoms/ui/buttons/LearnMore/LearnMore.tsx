import { IPrimaryProps, Primary } from "../Primary";

export interface ILearnMoreProps extends IPrimaryProps {
  textContent: string;
}

export function LearnMore(props: ILearnMoreProps) {
  const { textContent, ...rest } = props;

  const onClick = () => console.log("LearnMoreButton click");

  return (
    <Primary
      data-testid="LearnMore"
      textContent={textContent}
      onClick={onClick}
      {...rest}
    />
  );
}
