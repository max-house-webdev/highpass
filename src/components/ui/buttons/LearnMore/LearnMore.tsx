import { ButtonProps } from "@chakra-ui/react";

import { useEvent } from "@mh-utils/hooks/useEvent";

import { Primary } from "../Primary";

export interface ILearnMoreProps extends ButtonProps {
  textContent: string;
}

export function LearnMore(props: ILearnMoreProps) {
  const { textContent, ...rest } = props;

  const onClick = useEvent(() => console.log("LearnMoreButton click"));

  return (
    <Primary
      data-testid="LearnMore"
      textContent={textContent}
      onClick={onClick}
      {...rest}
    />
  );
}
