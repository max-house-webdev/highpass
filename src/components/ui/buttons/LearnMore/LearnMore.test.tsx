import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { LearnMore } from "./LearnMore";

describe("<LearnMore />", () => {
  test("it should mount", () => {
    render(<LearnMore />);

    const learnMore = screen.getByTestId("LearnMore");

    expect(learnMore).toBeInTheDocument();
  });
});
