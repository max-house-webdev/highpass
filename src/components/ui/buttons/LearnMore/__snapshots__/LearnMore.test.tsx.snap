// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LearnMore /> renders correctly 1`] = `
<button
  className="chakra-button css-4xx2wk"
  data-testid="LearnMore"
  onClick={[Function]}
  type="button"
>
  learn more
</button>
`;
